packages/edit-site/src/store/actions.js

```diff
@@ -43,7 +43,19 @@ export function __experimentalSetPreviewDeviceType( deviceType ) {
  *
  * @return {Object} Action object.
  */
-export function setTemplate( templateId ) {
+export function* setTemplate( templateId ) {
+	if ( Number.isNaN( Number( templateId ) ) ) {
+		templateId = yield findTemplate( '', templateId, [ templateId ] );
+
+		yield dispatch(
+			'core/data',
+			'invalidateResolution',
+			'core',
+			'getEntityRecords',
+			[ 'postType', 'wp_template' ]
+		);
+	}
```

Comment:
Mind adding some inline comments to explain what happens here? 🙂

Reply:
Actually, this `invalidateResolution` is temporary here. I have no idea where to put this to make more sense.

This function (`setTemplate`) creates an auto-draft if we pass a slug instead of an ID. `invalidateResolution` requires all the args which makes this invalidation very specific. Is there any way to invalidate all the `wp_template` queries? Docs didn't help :(